### DISCUSSION TITLE
STM32F4 I2s On different Pins

### DISCUSSION THREAD

**Discussion by mrwright16** [pschatzmann/stm32f411-i2s#4] [Q&A] [answered]
Is there a way to allow me to use SPI 2 for the I2S Instance? I have my blackpill wired up to use pins PB12-WS, PB13-BCK, and PB15 as data out to an amp. Im using the blackpoll to control and determine which folders to retrieve audio from an SD Breakout board.

**Comment by pschatzmann** [comment, marked as answer]
On STM processors only some defined pins can be used for I2S. I assume that your pins can be used for your processor variant.

The pins are defined in the [stm32-config-i21.h](https://github.com/pschatzmann/stm32f411-i2s/blob/main/src/stm32-config-i2s.h) configuration file. 

Feel free to edit your local file...

**Reply by mrwright16** [reply]
I was able to initiate it in CubeMX, so that's why I chose those pins to make wiring my circuit board a little simpler since the pins were grouped close together. I'll give changing the pin names a try. Would I have to change the I2S3 to I2S2? Also, by pin names in the config file there -  "{mclk, PB_10, GPIO_AF6_SPI3}," - Would I have to change GPIO_AF6_SPI3 to anything else?

**Reply by pschatzmann** [reply]
Yes you would need to use the codes proposed by CubeMX

**Reply by pschatzmann** [reply]
I always only used it on SPI3. To support SPI2 you might need to extend the logic a bit..
